src/transport/GroupPeerMessageCounter.cpp

```diff
@@ -277,9 +278,8 @@ CHIP_ERROR GroupOutgoingCounters::Init(chip::PersistentStorageDelegate * storage
     err = mStorage->SyncGetKeyValue(DefaultStorageKeyAllocator::GroupControlCounter().KeyName(), &temp, size);
     if (err == CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND)
     {
-        // might be the first time we retrieve the value
-        // TODO handle this case
-        mGroupControlCounter = 0; // TODO should be random
+        // First time retrieving the counter
+        mGroupControlCounter = chip::Crypto::GetRandU32();
```

Comment:
This does not match the spec.  The spec says:

> All message counters SHALL be initialized with a random value using the `Crypto_DRBG(len = 28) + 1` primitive.

See how the init in `src/transport/MessageCounter.h` is done.

Reply:
I guess the spec had time to change between the `todo` and the fix...

However it does look like the exact same thing except for the mask... 
https://github.com/project-chip/connectedhomeip/blob/master/src/transport/MessageCounter.h#L57